### PR TITLE
Settings refactor

### PR DESCRIPTION
**Pull request by KayaDLX**
# Refactor Settings Menu

- Now using ClientSettings cookies (taken from https://github.com/Facepunch/sbox-boomer)
- Saves settings across game sessions client's cookies 
- Minor main menu styling changes
- First support for common settings
- Start implementing localization files

Settings are defined in [GameSettings.cs](https://github.com/AmperSoftware/TF-Source-2/blob/settings-refactor/code/GameSettings.cs)

### CODE REVIEW

**Comment by KayaDLX** [comment]
I'm gonna move ClientSettings to Amper SDK because what I'm doing with some settings doesn't make any sense.

**Comment by KayaDLX** [comment]
Made it so it overrides SDK functions instead of forcing them down the SDK in an ugly way.

File: code/UI/HUD/DamageNumbers.cs [hunk 1]
```diff
@@ -37,9 +37,9 @@ void OnHurt( PlayerHurtEvent args )
 		// Play hit sound, if the player is alive.
 		if ( victim.IsAlive() )
 		{
-			if ( TimeSinceDing > 0.1f )
+			if ( TimeSinceDing > 0.1f && TFClientSettings.Current.PlayHitSound )
```
Comment (rob5300) [inline code/UI/HUD/DamageNumbers.cs:40]: Reverse order so `TFClientSettings.Current.PlayHitSound` is first.

**Review by rob5300 — CHANGES_REQUESTED** [review]
Small change then LGTM

**Review by rob5300 — APPROVED** [review]

**Pull request merged**